src/main/java/hudson/plugins/mercurial/MercurialStatus.java

```diff
@@ -84,7 +84,12 @@ private HttpResponse handleNotifyCommit(URI url) throws ServletException, IOExce
         final List<Item> projects = Lists.newArrayList();
         boolean scmFound = false,
                 urlFound = false;
-        for (Item project : Jenkins.getInstance().getAllItems()) {
+        final Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
+            return HttpResponses.error(SC_NOT_FOUND, "Jenkins instance is not ready");
```

Comment:
Probably a 503 would be more appropriate here. (Anyway web methods will not be serviced under this condition so `getActiveInstance` in 1.590+ would be more appropriate.)